src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,135 @@
+package org.jenkinsci.plugins.ewm.casc;
+
+import hudson.ExtensionList;
+import hudson.model.Computer;
+import hudson.model.Node;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.nodes.ExternalWorkspaceProperty;
+import org.jenkinsci.plugins.ewm.nodes.NodeDiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep;
+import org.jenkinsci.plugins.ewm.steps.ExwsStep;
+import org.jenkinsci.plugins.ewm.definitions.Template;
+import org.jenkinsci.plugins.ewm.nodes.NodeDisk;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.snakeyaml.Yaml;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test for Configuration As Code Compatibility.
+ *
+ * @author Martin d'Anjou
+ */
+public class ConfigAsCodeTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```

Comment:
You should NOT have to call this
use the ConfiguredWithCode annotation
https://github.com/jenkinsci/configuration-as-code-plugin/blob/09a6f0aae8b2a6ba16904a354e3ea377921161c9/plugin/src/test/java/io/jenkins/plugins/casc/JenkinsConfiguredWithCodeRuleClassRuleTest.java#L14-L16

Reply:
The ConfiguredWithCode annotation requires me to be able to import io.jenkins.plugins.casc.misc.ConfiguredWithCode I believe. The io.jenkins.plugins.casc.misc package is not accessible by loading the latest plugin as a dependency.

Reply:
Not sure why, but I think in order to be able to use that annotation, you have to define the `configuration-as-code` dependency twice in the `pom.xml`, an additional one with `classifier: tests`, like this: 

```
        <dependency>
            <groupId>io.jenkins</groupId>
            <artifactId>configuration-as-code</artifactId>
            <version>${jcasc.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.jenkins</groupId>
            <artifactId>configuration-as-code</artifactId>
            <version>${jcasc.version}</version>
            <classifier>tests</classifier>
            <scope>test</scope>
        </dependency>
```
I've seen this on other plugins e.g. https://github.com/jenkinsci/ec2-plugin/blob/master/pom.xml

I don't know if this is a best practice or not.

Reply:
This does work to give access to the ConfiguredWithCode annotation...unless I'm informed there is a method of better practice I will use this. Thanks for the advice here.

Reply:
All of this is documented in our JCasC docs folder :sweat_smile: 
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#how-to-test

Reply:
Terrific, thank you for pointing me to this. :)

Reply:
https://github.com/imaffe/external-workspace-manager-plugin/pull/5